freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/FreeIpaPasswordValidator.java

```diff
@@ -0,0 +1,41 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.freeipa.controller.exception.BadRequestException;
+
+@Component
+class FreeIpaPasswordValidator {
+
+    private static final Set<String> PATTERNS = Set.of("(?=.*[a-z]).*$", "(?=.*[A-Z]).*$", "(?=.*[0-9]).*$", ".*\\W+.*");
```

Comment:
This should match with : https://github.infra.cloudera.com/thunderhead/thunderhead/blob/master/services/api/environments2/src/main/java/com/cloudera/thunderhead/service/environments2api/server/PasswordValidator.java#L16

Reply:
The policy is based on JIRA https://jira.cloudera.com/browse/CB-2339. Can you please update the jira so we can make the proper change here.